cedar-policy-validator/src/human_schema/test.rs

```diff
@@ -1267,4 +1268,158 @@ mod translator_tests {
                 )
         );
     }
+
+    #[test]
+    fn entity_named_namespace() {
+        let src = r#"
+        entity namespace = {};
+        entity Foo in [namespace] = {};
+        "#;
+
+        let (schema, _) = SchemaFragment::from_str_natural(src).unwrap();
+        let ns = schema.0.get("").unwrap();
+        let foo = ns.entity_types.get("Foo").unwrap();
+        assert_eq!(foo.member_of_types, vec!["namespace".to_smolstr()]);
+    }
+
+    #[test]
+    fn entity_named_in() {
+        // This fails because `in` is reserved
```

Comment:
I don't think it should fail. `in` is a valid entity type name in Cedar.

Reply:
It is not an valid entity type name. Try it

Reply:
If we accept https://github.com/cedar-policy/rfcs/pull/48, then `id` should be a valid annotation (see https://github.com/cedar-policy/cedar/pull/634), so this fix is not ideal.

It is also an minor difference between the schema and policy parser. The policy parser will construct a CST containing invalid IDs and reject only when building a CST, but the schema parser is raising an error while parsing.